medpy/filter/smoothing.py

```diff
@@ -123,6 +130,12 @@ def condgradient(delta, spacing):
     elif option == 2:
         def condgradient(delta, spacing):
             return 1./(1.+(delta/kappa)**2.)/float(spacing)
+    elif option == 3:
+        kappa_s = kappa * (2**0.5)
+
+        def condgradient(delta, spacing):
+            top = 0.5/((1.-(delta/kappa_s)**2.)**2.)/float(spacing)
+            return numpy.where(delta <= kappa_s, top, 0)
```

Comment:
I think there should be an abs on the delta here?

Reply:
I would think so, too. The problem didn't arise with the Perona-Malik diffusion equations and the Tsiotsios et al. work does not address the problem. Could you fix this? Then I'll approve the pull.

Thanks a lot for the addition!

Reply:
[This](https://github.com/shirley9073/mipav/blob/3716b62b4715718424f5be92d10e70876822c0aa/src/gov/nih/mipav/model/algorithms/AlgorithmFastMarching.java#L27) seems to indicate using the abs